### ISSUE TITLE
Single line for/when shouldn't warn about ensure comprehensions

### ISSUE THREAD

**Issue by differentmatt** [clutchski/coffeelint#482]
```coffeescript
# Linter warning
instanceID = index for val, index in @instances when val.name is instanceName

# No linter warning
for val, index in @instances when val.name is instanceName
  instanceID = index
```

Related to #237

**Comment by swang** [comment]
Yes, ensure_comprehensions was set out to protect against this. Usually most users want a comprehension of the list, (a list from a list), and not just one element from the list. So the rule works as is intended, with your second piece of code you wrote as the workaround (besides just turning off the rule).

Obviously it is a bit clunky and not optimal, but there's no way to really figure out intent by the coder.
